frontend/src/pages/groupSettings/GroupSettings.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { useEffect, useState } from 'react';
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  PageSection,
+  PageSectionVariants,
+  Text,
+  HelperText,
+  HelperTextItem,
+  Alert,
+  AlertActionCloseButton,
+} from '@patternfly/react-core';
+import ApplicationsPage from '../ApplicationsPage';
+import { useDispatch } from 'react-redux';
+import { addNotification, detectUser } from '../../redux/actions/actions';
+import './GroupSettings.scss';
+import { fetchGroupsSettings, updateGroupsSettings } from 'services/groupSettingsService';
+import { GroupsConfig, GroupsConfigField, MenuItemStatus } from 'types';
+import { MenuOptionMultiSelect } from 'components/MultiSelection';
+
+const TITLE = `User and group settings`;
+const DESCRIPTION = `Define OpenShift group membership for Data Science administrators and users.`;
+const ERROR_MESSAGE = `Unable to load User and group settings`;
+const EMPTY_MESSAGE = `No user and group settings found`;
+const ADMIN_FORM_TITLE = `Data Science administrator groups`;
+const ADMIN_FORM_DESCRIPTION = `Select the OpenShift groups that contain all Data Science administrators.`;
+const USER_FORM_TITLE = `Data Science user groups`;
+const USER_FORM_DESCRIPTION = `Select the OpenShift groups that contain all Data Science users.`;
+const CARD_FOOTER = `View, edit, or create groups in OpenShift under User Management`;
+const BUTTON_LABEL = `Save changes`;
+const ALERT_TITLE = `Group no longer exists`;
```

Comment:
mmmm I can argue this. when you think about localization , it's fairly common to have keys in the inline string rendering, some libraries such as react-i18next force you to have its hook with the attribute, and it's fairly common to have all the text localized with descriptive names. Also, it's useful when you need to reuse some texts (such as in this case). I'm gonna move the text to the inline render but keep the ones that are repeated (because performance wise calling a variable in each render is not that expensive). Thanks again Andrew!